File: examples/python/snippets/llm-proxy-anthropic-sdk.py
import os
from anthropic import Anthropic

default_headers = {}
default_headers["Authorization"] = f"Bearer {os.environ['MESHAGENT_ACCESS_TOKEN']}"
project_id = os.environ.get("MESHAGENT_PROJECT_ID")
if project_id:
    default_headers["Meshagent-Project-Id"] = project_id

client = Anthropic(
    base_url=os.environ.get(
        "MESHAGENT_ANTHROPIC_BASE_URL",
        "https://api.meshagent.com/anthropic",
    ),
    api_key="meshagent",
    default_headers=default_headers,
)

message = client.messages.create(
    model="claude-sonnet-4-6",
    max_tokens=512,
    messages=[
        {
            "role": "user",
            "content": "Tell me a fun fact about AI.",
        }
    ],
)

print(message.content[0].text)

File: examples/typescript/llm-proxy-anthropic-sdk.ts
import Anthropic from "@anthropic-ai/sdk";

async function main() {
  const defaultHeaders: Record<string, string> = {
    Authorization: `Bearer ${process.env.MESHAGENT_ACCESS_TOKEN!}`,
  };
  if (process.env.MESHAGENT_PROJECT_ID) {
    defaultHeaders["Meshagent-Project-Id"] = process.env.MESHAGENT_PROJECT_ID;
  }

  const client = new Anthropic({
    baseURL:
      process.env.MESHAGENT_ANTHROPIC_BASE_URL ??
      "https://api.meshagent.com/anthropic",
    apiKey: "meshagent",
    defaultHeaders,
  });

  const message = await client.messages.create({
    model: "claude-sonnet-4-6",
    max_tokens: 512,
    messages: [
      {
        role: "user",
        content: "Tell me a fun fact about AI.",
      },
    ],
  });

  console.log(message.content[0]);
}

void main();
